src/interaction/Simulation.cpp

```diff
@@ -0,0 +1,189 @@
+}
+
+void Simulation::create_animations()
+{
+  if (m_simulation_states.size() > 1)
+  {
+    SimulationState first_position = m_simulation_states.front();
+    ch::Sequence<glm::vec3> sequence_eye(first_position.poi.eye);
+    ch::Sequence<glm::vec3> sequence_target(first_position.poi.target);
+    ch::Sequence<glm::vec3> sequence_up(first_position.poi.up);
+    ch::Sequence<float> sequence_radius(first_position.poi.radius);
+    ch::Sequence<glm::vec3> sequence_clip_max(first_position.max_clip);
+    ch::Sequence<glm::vec3> sequence_clip_min(first_position.min_clip);
+
+    for (auto iterator = std::next(m_simulation_states.begin()); iterator != m_simulation_states.end(); iterator++)
+    {
+      sequence_eye.then<ch::RampTo>(iterator->poi.eye, m_animation_duration);
+      sequence_target.then<ch::RampTo>(iterator->poi.target, m_animation_duration);
+      sequence_up.then<ch::RampTo>(iterator->poi.up, m_animation_duration);
+      sequence_radius.then<ch::RampTo>(iterator->poi.radius, m_animation_duration);
+      sequence_clip_max.then<ch::RampTo>(iterator->max_clip, m_animation_duration);
+      sequence_clip_min.then<ch::RampTo>(iterator->min_clip, m_animation_duration);
+    }
+
+
+    auto group = std::make_shared<ch::Timeline>();
+    group->apply<glm::vec3>(&m_eye_animation, sequence_eye);
+    group->apply<glm::vec3>(&m_target_animation, sequence_target);
+    group->apply<glm::vec3>(&m_up_animation, sequence_up);
+    group->apply<float>(&m_radius_animation, sequence_radius);
+    group->apply<glm::vec3>(&m_max_clip_animation, sequence_clip_max);
+    group->apply<glm::vec3>(&m_min_clip_animation, sequence_clip_min);
+
+    m_timeline.addShared(group);
+
+  }
+}
+
+void Simulation::update_simulation()
+{
+  if (m_animation_state != PAUSE)
+  {
+    update_time_step();
+
+  }
+  if (m_timeline.isFinished())
+  {
+    m_animation_state = STOP;
+    animation_button_label = "Animate";
+    m_controller_app.set_app_mode(MANUAL);
+    m_timeline.resetTime();
+    if (m_controller_app.get_movie_state() == MOVIE_RECORD)
+    {
+      m_controller_app.stop_movie();
+    }
+  }
+
+}
+
+void Simulation::update_time_step()
+{
+  m_timeline.step(1.0 / 30.0);
+  
+}
+
+
+void Simulation::set_animation_state()
+{
+  if (m_animation_state == STOP)
+  {
+    create_animations();
+    m_controller_app.set_app_mode(SIMULATION);
+    m_animation_state = PLAYING;
+    animation_button_label = "PAUSE";
+  }
+  else if (m_animation_state == PLAYING)
+  {
+    if (!m_timeline.isFinished())
+    {
+      m_animation_state = PAUSE;
+      animation_button_label = "ANIMATE";
+    }
+  }
+  else if (m_animation_state == PAUSE)
+  {
+    m_animation_state = PLAYING;
+    animation_button_label = "PAUSE";
+  }
+
+  m_is_animate_path = true;
+
+}
+
+ANIMATION_STATE Simulation::get_animation_state()
+{
+  if (m_timeline.isFinished())
+  {
+    return STOP;
+  }
+  else if (!m_timeline.isFinished() && m_animation_state == PLAYING)
```

Comment:
Should this be a switch statement?

Reply:
I replace the ifs to switch as suggested.